lib/zlib.js

```diff
@@ -146,7 +146,14 @@ function zlibOnError(message, errno) {
   var error = new Error(message);
   error.errno = errno;
   error.code = codes[errno];
-  this.emit('error', error);
+
+  if (this._initSuccess) {
+    this.emit('error', error);
+  } else {
+    process.nextTick(() => {
+      this.emit('error', error);
+    });
+  }
```

Comment:
You could change the `ZCtx::Error(ctx, "Init error");` line to throw an error instead … that should work, right?